bitnami/grafana-alloy/templates/cluster_service.yaml

```
@@ -0,0 +1,47 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if .Values.alloy.clustering.enabled -}}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ printf "%s-cluster" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: alloy
+    app.kubernetes.io/part-of: grafana-alloy
+  {{- if or .Values.service.annotations .Values.commonAnnotations }}
+  {{- $annotations := include "common.tplvalues.merge" (dict "values" (list .Values.service.annotations .Values.commonAnnotations) "context" .) }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  type: ClusterIP
+  clusterIP: {{ .Values.service.clusterIP }}
+  {{- if .Values.service.sessionAffinity }}
+  sessionAffinity: {{ .Values.service.sessionAffinity }}
+  {{- end }}
+  {{- if .Values.service.sessionAffinityConfig }}
+  sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" .Values.service.sessionAffinityConfig "context" $) | nindent 4 }}
+  {{- end }}
+  ports:
+    # This service should only be used for clustering, and not metric
+    # collection.
+    - name: http
+      port: {{ .Values.alloy.containerPorts.http }}
+      targetPort: {{ .Values.alloy.containerPorts.http }}
```

Name ports are preferred for "targetPort":
```suggestion
      targetPort: http
```